tests/scripts/thread-cert/Cert_7_1_03_BorderRouterAsLeader.py

```diff
@@ -96,91 +118,123 @@ def test(self):
-            sent_to_node=self.nodes[SED1],
-        )
-        check_child_update_response(msg, address_registration=CheckType.CONTAIN, CIDs=[1])
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        ROUTER = pv.vars['ROUTER']
+        SED = pv.vars['SED']
+        MED = pv.vars['MED']
+
+        # Step 1: Ensure topology is formed correctly
+        pv.verify_attached('ROUTER', 'LEADER')
+        pv.verify_attached('SED', 'LEADER', 'MTD')
+        pv.verify_attached('MED', 'LEADER', 'MTD')
+
+        # Step 3: The DUT MUST send a multicast MLE Data Response,
+        #         including the following TLVs:
+        #             - Network Data TLV
+        #               At least two Prefix TLVs (Prefix 1 and Prefix 2),
+        #               each including:
+        #                   - 6LoWPAN ID sub-TLV
+        #                   - Border Router sub-TLV
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_DATA_RESPONSE).\
+            filter(lambda p: {
+                              Ipv6Addr(PREFIX_2001[:-3]),
+                              Ipv6Addr(PREFIX_2002[:-3])
+                             } == set(p.thread_nwd.tlv.prefix) and\
+                   p.thread_nwd.tlv.border_router.flag.p == [1, 1] and\
+                   p.thread_nwd.tlv.border_router.flag.s == [1, 1] and\
+                   p.thread_nwd.tlv.border_router.flag.r == [1, 1] and\
+                   p.thread_nwd.tlv.border_router.flag.o == [1, 1] and\
+                   p.thread_nwd.tlv.stable == [0, 1, 1, 1, 0, 0, 0]
+                   ).\
+            must_next()
+
+        # Step 4: MED automatically sends the global address configured to its parent
+        #         (the DUT), via the Address Registration TLV included in its Child
+        #         Update Request keep-alive message.
+
+        # Step 5: The DUT MUST send a MLE Child Update Response to MED
+        #          The following TLVs MUST be present in the Child Update Response:
+        #              - Source Address TLV
+        #              - Address Registration TLV
+        #                  - Echoes back addresses configured in step 4
+        #              - Mode TLV
+        with pkts.save_index():
+            _pkt = pkts.filter_wpan_src64(MED).\
+                filter_wpan_dst64(LEADER).\
+                filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).\
+                must_next()
+            pkts.filter_wpan_src64(LEADER).\
+                filter_wpan_dst64(MED).\
+                filter_mle_cmd(MLE_CHILD_UPDATE_RESPONSE).\
+                filter(lambda p: {
+                                  SOURCE_ADDRESS_TLV,
+                                  MODE_TLV,
+                                  ADDRESS_REGISTRATION_TLV
+                                 } < set(p.mle.tlv.type) and\
+                       set(p.mle.tlv.addr_reg_iid) < set(_pkt.mle.tlv.addr_reg_iid)
```

Comment:
further validate the addresses echoed back?
The current check can pass even when no address is echoed back.

Reply:
added nullField check